### ISSUE TITLE
To-do

### ISSUE THREAD

**Issue by muellerdo** [frankkramer-lab/riadd.aucmedi#3]
- [x] Implement directory interface in AUCMEDI
- [x] Implement inference augmenting in AUCMEDI
- [x] Implement augmenting predictions in baseline as well for comparison with normal baseline prediction
- [x] Run baseline again
- [x] Evaluate baseline
- [x] Compute baseline predictions for evaluation set with some model (upload and see score on online evaluation set)
- [x] Implement up-sampling approach for rare label combinations
- [x] Run classifier 5-fold CV with selected architectures
- [x] Implement ensembler & ensembler Training
- [x] Run ensembler training
- [x] Implement prediction
- [x] Test out single model for each class ML algorithms
- [x] Implement sample weights for RF label approach
- [x] Implement disease risk binary model and train it
- [x] Implement focal loss to AUCMEDI
- [x] Implement multi-label class weights in AUCMEDI
- [x] Implement ensemble learning for disease risk
- [x] Implement single architecture inference for Focal Loss EfficientNetB4 and upload
- [x] Implement focal loss usage to all models

**Further:**
- [x] Test out Single Model Learning for each class Ensemble with Random Forest
- [x] Implement disease risk to ensembling
- [x] Project structuring
- [x] Try out longer transfer learning epochs (20?) with lower learning rate (E-07?)
- [x] Increase epochs to 300
- [x] Try out focal loss https://github.com/sushanttripathy/Keras_loss_functions/blob/master/focal_loss.py
https://stackoverflow.com/questions/48485870/multi-label-classification-with-class-weights-in-keras

**Final:**  
- [x] Implement Evaluation
- [x] Re-run all classifiers and detector trainings
- [x] Re-run ensemble training
- [x] Re-run inferences
- [x] Re-run ensemble
- [x] Upload 
- [x] Run Evaluation on final data
- [x] Write Manuscript
- [x] Revise Manuscript
- [x] Update README
- [x] Upload Data to Zendo
- [x] Make this repo public

**Upload:**
- [x] Simple LR approach
- [x] Simple RF approach
- [x] Augmenting LR approach
- [x] Augmenting RF approach

**Idea:**
- Run predictions on training dataset (detector & label)
- Collect all predictions  (detector & label)
- Train a LR/RF Model for each class
